libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/EHorizonObjectMetadata.kt

```diff
@@ -9,11 +11,24 @@ import com.mapbox.navigation.base.trip.model.alert.IncidentInfo
  * @param objectProvider provider of road object
  * @param incidentInfo will be filled only if [type] is [EHorizonObjectType.INCIDENT] and
  * [objectProvider] is [EHorizonObjectProvider.MAPBOX]
+ * @param tunnelInfo will be filled only if [type] is [EHorizonObjectType.TUNNEL_ENTRANCE] and
+ * [objectProvider] is [EHorizonObjectProvider.MAPBOX]
+ * @param borderCrossingInfo will be filled only if [type] is [EHorizonObjectType.BORDER_CROSSING]
+ * and [objectProvider] is [EHorizonObjectProvider.MAPBOX]
+ * @param tollCollectionType will be filled only if [type] is
+ * [EHorizonObjectType.TOLL_COLLECTION_POINT] and [objectProvider] is
+ * [EHorizonObjectProvider.MAPBOX]
+ * @param restStopType will be filled only if [type] is [EHorizonObjectType.SERVICE_AREA] and
+ * 42.[objectProvider] is [EHorizonObjectProvider.MAPBOX]
  */
 class EHorizonObjectMetadata internal constructor(
```

Comment:
Should we add a `EHorizonObjectMetadata` `BuilderTest`? Noticed that we didn't add these for the builders in `trip.model.eh`, was that on purpose?

Reply:
answered below